File: 2024 styczeń/INF.04-01-24.01-SG/Aplikacja Konsolowa/C++/main.cpp
#include <iostream>

using namespace std;

char sexFinder(int PESEL[])
{
    int sexNumberPosition = PESEL[9];
    if (sexNumberPosition % 2 == 0)
    {
        return 'K';
    }
    else
    {
        return 'M';
    }
}

bool controlSum(int PESEL[])
{
    int sum = 0;
    int j = 1;
    for (int i = 0; i < 10; i++)
    {
        if (j == 1)
        {
            sum += PESEL[i] * 1;
        }
        else if (j == 2)
        {
            sum += PESEL[i] * 3;
        }
        else if (j == 3)
        {
            sum += PESEL[i] * 7;
        }
        else if (j == 4)
        {
            sum += PESEL[i] * 9;
            j = 0;
        }
        j++;
    }

    int m = sum % 10;
    int r;

    if (m == 0)
    {
        r = 0;
    }
    else
    {
        r = 10 - m;
    }

    if (r == PESEL[10])
    {
        return true;
    }
    else
    {
        return false;
    }
}

int main()
{
    string pesel;
    cout << "Podaj pesel: ";
    cin >> pesel;

    int peselArray[11];

    for (int i = 0; i < pesel.length(); i++)
    {
        peselArray[i] = pesel.at(i) - '0';
    }

    char sex = sexFinder(peselArray);
    cout << "Płeć: " << sex << endl;

    bool sum = controlSum(peselArray);
    if (sum)
    {
        cout << "Suma kontrolna poprawna!";
    }
    else
    {
        cout << "Suma kontrolna niepoprawna!";
    }

    return 0;
}

File: 2024 styczeń/INF.04-01-24.01-SG/Aplikacja Konsolowa/Python/main.py
"""
    nazwa funkcji: sexFinder
    opis funkcji: funckja sprawdza czy podany pesel należy do kobiety czy do mężczyzny
    parametry: PESEL - pesel do sprawdzenia
    zwracany typ i opis: fukcja zwraca znak "K" dla kobiety lub znak "M" dla mężczyzny
    autor: 00000000000
"""
def sexFinder(PESEL):
    sexNumberPosition = int(PESEL[9])
    if sexNumberPosition % 2 == 0:
        return "K"
    else:
        return "M"

def controlSum(PESEL):
    sum = 0
    j = 1
    for i in range(10):
        if j == 1:
            sum += int(PESEL[i]) * 1
        elif j == 2:
            sum += int(PESEL[i]) * 3
        elif j == 3:
            sum += int(PESEL[i]) * 7
        elif j == 4:
            sum += int(PESEL[i]) * 9
            j = 0
        j = j + 1

    m = sum % 10
    r = 0
    if m != 0:
        r = 10 - m
    if r == int(PESEL[10]):
        return True
    else:
        return False
    
pesel = input("Podaj pesel: ")
peselArray = []

sex = sexFinder(pesel)
print("Płeć:", sex)

sum = controlSum(pesel)
if sum:
    print("Suma kontrolna poprawna!")
else:
    print("Suma kontrolna niepoprawna!")
    